demo/openssl-qat-engine/Dockerfile

```diff
@@ -1,4 +1,31 @@
-FROM ubuntu:24.04
+FROM ubuntu:24.04 AS builder
+
+ARG QAT_ENGINE_VERSION="v1.6.2"
 
 RUN apt update && \
-    apt install --no-install-recommends -y qatengine qatlib-examples qatzip openssl
+    env DEBIAN_FRONTEND=noninteractive apt install -y \
+    git \
+    autoconf \
+    g++ \
+    pkg-config \
+    libssl-dev \
+    libtool \
+    make \
+    libusdm-dev
+    
+RUN git clone --depth 1 -b $QAT_ENGINE_VERSION https://github.com/intel/QAT_Engine
+
+RUN cd /QAT_Engine && \
+    sed -i '/fprintf(stderr, "QAT_HW device not available & QAT_SW not enabled. Using OpenSSL_SW!\\n");/ s/$/ return ret;/' e_qat.c && \
+    ./autogen.sh && \
+    ./configure \
+    --disable-qat_sw --with-qat_engine_id=qathwtest && \
+    make && make test && make install
+
+FROM ubuntu:24.04
+
+COPY --from=builder /QAT_Engine/testapp /usr/bin/
+COPY --from=builder /usr/bin/*_sample* /usr/bin/
+COPY --from=builder /usr/lib/x86_64-linux-gnu/engines-3/qatengine.so /usr/lib/x86_64-linux-gnu/engines-3/qathwtest.so
+COPY --from=builder /QAT_Engine/LICENSE /usr/share/package-licenses/QAT_Engine/LICENSE
+RUN ldconfig && apt update && env DEBIAN_FRONTEND=noninteractive apt install -y openssl libcrypto-mb-dev libqat4 libusdm-dev
```

Comment:
```suggestion
RUN apt update && env DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends -y qatengine qatlib-examples
```

Reply:
Ah, I didn't know you are also writing this.
Isn't `ldconfig` necessary?

Reply:
yeah, maybe not necessary. testapp works without that.

Reply:
Thanks! :)